src/core/vector/qgsvectorlayer.cpp

```diff
@@ -6109,22 +6109,21 @@ void QgsVectorLayer::emitDataChanged()
   if ( mDataChangedFired )
     return;
 
+  // If we are asked to fire dataChanged from a layer we depend on,
+  // be sure that this layer is not in the process of committing its changes, because
+  // we will be asked to fire dataChanged at the end of his commit, and we don't
+  // want to fire this signal more than necessary.
+  if ( QgsVectorLayer *layerWeDependUpon = dynamic_cast<QgsVectorLayer *>( sender() ) )
+    if ( layerWeDependUpon->mCommitChangesActive )
```

Comment:
```suggestion
  if ( ( QgsVectorLayer *layerWeDependUpon = qobject_cast<QgsVectorLayer *>( sender() ) ) && layerWeDependUpon->mCommitChangesActive )
```

one lining and qobject_cast is supposedly better than dynamic_cast when it comes to QObject